src/components/dashboard/main-content/course-enrollments/course-cards/CompletedCourseCard.jsx

```diff
@@ -1,36 +1,143 @@
+    endDate,
+  } = props;
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+
+  const handleUnarchiveOnClose = () => {
+    setIsModalOpen(false);
+    sendTrackEvent('edx.learner_portal.course.unarchive.modal.closed', {
+      course_run_id: courseRunId,
+    });
+  };
+
+  const handleUnarchiveOnSuccess = ({ response, resetModalState }) => {
+    sendTrackEvent('edx.learner_portal.course.unarchive.saved', {
+      course_run_id: courseRunId,
+    });
+    setIsModalOpen(false);
+    resetModalState();
+    modifyCourseRunStatus({
+      status: response.courseRunStatus,
+      courseId: response.courseRunId,
+      markedDone: response.markedDone.toLowerCase() === 'true',
+    });
+    modifyIsUnarchiveCourseStatus({
+      isSuccess: true,
+    });
+  };
+
+  const getDropdownMenuItems = () => {
+    // Only courses that are manually archived (markedDone)
+    // should show a unarchive option.
+    // if course is ended, you cannot unarchive it
+    if (!markedDone || courseEnded(endDate)) { return []; }
+    return ([
+      {
+        key: 'unarchive-course',
+        type: 'button',
+        onClick: () => {
+          setIsModalOpen(true);
+          sendTrackEvent('edx.learner_portal.course.unarchive.modal.opened', {
+            course_run_id: courseRunId,
+          });
+        },
+        children: (
+          <div role="menuitem">
+            Unarchive course
+            <span className="sr-only">for {title}</span>
           </div>
+        ),
+      },
+    ]);
+  };
+
+  const renderButtons = () => {
+    if (courseEnded(endDate)) { return null; }
+    return (
+      <ContinueLearningButton
+        linkToCourse={linkToCourse}
+        title={title}
+        courseRunId
+      />
+    );
+  };
+
+  const renderCertificateInfo = () => (
+    props.linkToCertificate ? (
+      <div className="d-flex mb-3">
+        <div className="mr-3">
+          <img src={CertificateImg} alt="verified certificate preview" />
+        </div>
+        <div className="d-flex align-items-center">
+          <p className="lead mb-0 font-weight-normal">
+            View your certificate on{' '}
+            <a
+              className="text-underline"
+              href={`${process.env.LMS_BASE_URL}/u/${username}`}
+            >
+              your profile →
+            </a>
+          </p>
         </div>
-      ) : (
-        <p className="lead mb-3 font-weight-normal">
-          To earn a certificate,
-          {' '}
-          <a className="text-underline" href={props.linkToCourse}>retake this course →</a>
-        </p>
-      )}
+      </div>
+    ) : (
+      <p className="lead mb-3 font-weight-normal">
+        To earn a certificate,{' '}
+        <a className="text-underline" href={props.linkToCourse}>
+          retake this course →
+        </a>
+      </p>
+    )
+  );
+
+  return (
+    <BaseCourseCard
+      buttons={renderButtons()}
+      dropdownMenuItems={getDropdownMenuItems({ markedDone })}
```

Comment:
nit: is this an unnecessary argument? doesn't look like `getDropdownMenuItems` takes any arguments.

Reply:
damn, you are right! I moved markdDone to the store already